UnitTests/GitUITests/FindFilePredicateProviderTest.cs

```diff
@@ -0,0 +1,36 @@
+﻿using GitUI;
+using NUnit.Framework;
+
+namespace GitUITests
+{
+    [TestFixture]
+    public class FindFilePredicateProviderTest
+    {
+        private IFindFilePredicateProvider provider;
+
+        [SetUp]
+        public void Init()
+        {
+            provider = new FindFilePredicateProvider();
+        }
+
+        [TestCase(@"Code", @"D:\", "TestCode", ExpectedResult = true)]
+        [TestCase(@"Code", @"D:\", "TestCode/File1", ExpectedResult = true)]
+        [TestCase(@"code", @"D:\", "TestCode/File1", ExpectedResult = true)]
+        [TestCase(@"/Code", @"D:\", "TestCode/File1", ExpectedResult = false)]
+        [TestCase(@"Code/", @"D:\", "TestCode/File1", ExpectedResult = true)]
+        [TestCase(@"Code\", @"D:\", "TestCode/File1", ExpectedResult = true)]
+        [TestCase(@"D:/Code\", @"D:\", "TestCode/File1", ExpectedResult = false)]
+        [TestCase(@"D:/Test/", @"D:\", "TestCode/File1", ExpectedResult = false)]
+        [TestCase(@"D:\Test", @"D:\", "TestCode/File1", ExpectedResult = true)]
+        [TestCase(@"//d/git\Test", @"//d/git", "TestCode/File1", ExpectedResult = true)]
+        [TestCase(@"c/dir1/dir2", @"F:\", "src/dir1/dir2/dir3", ExpectedResult = true)]
+        [TestCase(@"c\dir1/dir2", @"F:\", "src/dir1/dir2/dir3", ExpectedResult = true)]
+        [TestCase(@"c/dir1\dir2", @"F:\", "src/dir1/dir2/dir3", ExpectedResult = true)]
+        public bool UsageScenario(string pattern, string workingDir, string fileName)
```

Comment:
I find it hard to reason about these tests.
Could you perhaps break them up per code path.
Something like:

* Get_should_throw_if_name_null_or_empty
* Get_should_throw_if_workingDir_null_or_empty
* Get_should_return_???_if_<name>_starts_with_workingDir

and so on